### ISSUE TITLE
/checkout/payment.js.coffee ajax error not preventing form submit

### ISSUE THREAD

**Issue by Jincheng-Li** [spree/spree#6561]
I'm seeing this issue on plain spree 2.2.11, but the code remains unchanged in master and I guess all versions have the same issue.

Setup
- spree 2.2.11 with Stripe gateway

Steps to reproduce
1. add a product to cart and step checkout to payment
2. choose credit card payment method and enter credit card info
2. enter an invalid coupon code and hit continue button
3. form is submitted and credit card info entered is lost

Expected behavior:
form submission is prevented if coupon is invalid

Suspected cause:
$.ajax's return value is not the function in success or error. Instead $.ajax always returns something that's will be evaluated as true.

Code from /checkout/payment.js.coffee

``` coffeescript
coupon_status.removeClass();
          $.ajax({
            async: false,
            method: "PUT",
            url: url,
            success: (data) ->
              coupon_code_field.val('')
              coupon_status.addClass("alert-success").html("Coupon code applied successfully.")
              return true
            error: (xhr) ->
              handler = JSON.parse(xhr.responseText)
              coupon_status.addClass("alert-error").html(handler["error"])
              $('.continue').attr('disabled', false)
              return false
          })
```

**Comment by priyank-gupta** [comment]
I can replicate this. On it.

**Comment by bbonislawski** [comment]
Fixed by #7040
